self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryInvoicesTestCase.java

```diff
@@ -134,33 +138,153 @@ public void paysInvoice(){
 
         MatcherAssert.assertThat("Invoice should be paid",
-            storage.invoices().isPaid(1));
+            storage.invoices().isPaid(1),
+            Matchers.is(true));
+        MatcherAssert.assertThat("Invoice of contract should be paid",
+            invoicesOfContract.isPaid(1),
+            Matchers.is(true));
         MatcherAssert.assertThat(
-            "It should be to invoices one paid and o active",
-            contract.invoices(), Matchers.iterableWithSize(2));
+            "It should be two invoices one paid and one active",
+            contract.invoices(),
+            Matchers.iterableWithSize(2));
+
+    }
+
+    /**
+     * Throws IllegalStateException if task is not assigned.
+     */
+    @Test(expected = IllegalStateException.class)
+    public void throwsIfTaskNotAssigned(){
+        final Storage storage = new InMemory();
+        storage.invoices().add(Mockito.mock(Task.class), Duration.ZERO);
+    }
+
+    /**
+     * Throws IllegalStateException if paid invoice is not part of contract.
+     */
+    @Test(expected = IllegalStateException.class)
+    public void throwsIfPaidInvoiceIsNotPartOfContract(){
+        final Storage storage = new InMemory();
+        final ProjectManager projectManager = storage.projectManagers()
+            .pick(Provider.Names.GITHUB);
+        final Project project = storage
+            .projects()
+            .register(
+                this.mockRepo("john/test", Provider.Names.GITHUB),
+                projectManager
+            );
+        final Contributor contributor = storage
+            .contributors()
+            .register("mihai", Provider.Names.GITHUB);
+        //adding two contracts
+        storage.contracts().addContract(
+            project.repoFullName(),
+            contributor.username(),
+            contributor.provider(),
+            BigDecimal.ONE,
+            Contract.Roles.DEV
+        );
+        storage.contracts().addContract(
+            project.repoFullName(),
+            contributor.username(),
+            contributor.provider(),
+            BigDecimal.ONE,
+            Contract.Roles.ARCH
+        );
+        final Contract.Id contractIdOne = new Contract.Id(
+            project.repoFullName(),
+            contributor.username(),
+            contributor.provider(),
+            Contract.Roles.DEV
+        );
+        final Contract.Id contractIdTwo = new Contract.Id(
+            project.repoFullName(),
+            contributor.username(),
+            contributor.provider(),
+            Contract.Roles.ARCH
+        );
+        storage.invoices().add(this.mockAssignedTask(
+            "123", storage, project, contributor, Contract.Roles.DEV),
+            Duration.ofHours(1));
+        storage.invoices().add(this.mockAssignedTask(
+            "124", storage, project, contributor, Contract.Roles.ARCH),
+            Duration.ofHours(2));
 
+        storage.invoices().ofContract(contractIdTwo).isPaid(1);
```

Comment:
@criske What is the ``1`` argument in ``.isPaid(1)``? Is it the InvoiceId?

Reply:
Yes